### PR TITLE
adding pagination and pagination tests

### PR DESCRIPTION
**Pull request by pmhansen8**
Added pagination where users can query players by page size and page number. In the future we could possibly add filters including glicko, school, name, and more.

### CODE REVIEW

File: src/main/java/com/example/botfightwebserver/player/PlayerService.java [hunk 1]
```diff
@@ -101,6 +104,24 @@ public boolean setCurrentSubmissionIfNone(Long playerId, Long submissionId) {
...
+        }
+        if (size <= 0) {
+            throw new IllegalArgumentException("Page size must be greater than 0");
+        }
+
+        Pageable pageable = PageRequest.of(page, size);
+
+        // Fetch the players for the specified page
+        Page<Player> playerPage = playerRepository.findAll(pageable);
+
```
Comment (Benler123) [inline src/main/java/com/example/botfightwebserver/player/PlayerService.java:127]: I should've mentioned this, but the players and pages should be sorted by their glicko rating. So Page 1 has the {size} best players sorted in order, then page 2 and so forth.

File: src/test/java/com/example/botfightwebserver/player/PlayerServiceTest.java [hunk 2]
```diff
@@ -465,5 +466,87 @@ void testGetCurrentSubmission_NoSubmission() {
...
+
+    @Test
+    void testPagination_InvalidPageOrSize() {
+        // Test invalid page size
+        IllegalArgumentException exceptionForInvalidSize = assertThrows(
+                IllegalArgumentException.class,
+                () -> playerService.pagination(0, 0)
+        );
+        assertEquals("Page size must be greater than 0", exceptionForInvalidSize.getMessage());
+
```
Comment (Benler123) [inline src/test/java/com/example/botfightwebserver/player/PlayerServiceTest.java:526]: Tests look good!

File: src/main/java/com/example/botfightwebserver/player/PlayerService.java [hunk 3]
```diff
@@ -101,6 +104,24 @@ public boolean setCurrentSubmissionIfNone(Long playerId, Long submissionId) {
...
+        // Return the players as an array
+        return playerPage.getContent().toArray(new Player[0]);
```
Comment (Benler123) [inline src/main/java/com/example/botfightwebserver/player/PlayerService.java]: Prefer List over array

File: src/main/java/com/example/botfightwebserver/player/PlayerService.java (hunk 1 above)
Comment (pmhansen8) [reply src/main/java/com/example/botfightwebserver/player/PlayerService.java:127]: Added!

File: src/main/java/com/example/botfightwebserver/player/PlayerService.java (hunk 3 above)
Comment (pmhansen8) [reply src/main/java/com/example/botfightwebserver/player/PlayerService.java]: Changed!

File: src/test/java/com/example/botfightwebserver/player/PlayerServiceTest.java (hunk 2 above)
Comment (pmhansen8) [reply src/test/java/com/example/botfightwebserver/player/PlayerServiceTest.java:526]: Thank you🫦

**Review by Benler123 — APPROVED** [review]

**Pull request merged**